packages/ts-plugin/src/frontmatter.ts

```diff
@@ -0,0 +1,133 @@
+import { yaml2ts } from '@astrojs/yaml2ts';
+import {
+	type CodeMapping,
+	type LanguagePlugin,
+	type VirtualCode,
+	forEachEmbeddedCode,
+} from '@volar/language-core';
+import { pathToFileURL } from 'node:url';
+import type ts from 'typescript';
+
+const SUPPORTED_FRONTMATTER_EXTENSIONS = { md: 'markdown', mdx: 'mdx', mdoc: 'mdoc' };
+const SUPPORTED_FRONTMATTER_EXTENSIONS_KEYS = Object.keys(SUPPORTED_FRONTMATTER_EXTENSIONS);
+const SUPPORTED_FRONTMATTER_EXTENSIONS_VALUES = Object.values(SUPPORTED_FRONTMATTER_EXTENSIONS);
```

Comment:
This is repeated code, any chance to make this shared?

Reply:
It's a bit wonky to make shared because it's in different packages and it doesn't *really* belong in the YAML 2 TS package since it doesn't deal with the supported languages, but I guess it could be there